macros/edr/tests/test_all_columns_anomalies.sql

```diff
@@ -16,20 +16,25 @@
                                                                                    identifier=temp_metrics_table_name,
                                                                                    type='table') -%}
 
-        {#- get column configuration -#}
-        {%- set table_config = elementary.get_table_config_from_graph(model) %}
-        {{- elementary.debug_log('table config - ' ~ table_config) }}
+        {#- get all columns configuration -#}
         {%- set full_table_name = elementary.relation_to_full_name(model) %}
         {%- set model_relation = dbt.load_relation(model) %}
         {%- if not model_relation %}
             {{- elementary.test_log('monitored_table_not_found', full_table_name) }}
             {{- return(elementary.no_results_query()) }}
         {%- endif %}
 
-        {%- set timestamp_column = elementary.insensitive_get_dict_value(table_config, 'timestamp_column') %}
-        {{- elementary.debug_log('timestamp_column - ' ~ timestamp_column) }}
-        {%- set timestamp_column_data_type = elementary.insensitive_get_dict_value(table_config, 'timestamp_column_data_type') %}
-        {{- elementary.debug_log('timestamp_column_data_type - ' ~ timestamp_column_data_type) }}
+        {% if timestamp_column %}
```

Comment:
Looking at the code now I think we can write this code in a separate macro (also to avoid code duplication, I know it has been this way before but we can at least avoid adding more to it). Also, the whole purpose of get_table_config_from_graph is to extract the timestamp column. So I think we can rename it to get_timestamp_column_from_graph (the graph is dbt's object that holds all the metadata about the models) and if we don't have the timestamp_column (meaning, its value is none) call this get_timestamp_column_from_graph and then extrant the data type by calling your new function find_normalized_data_type_for_column.